apps/extension/src/core/constants.ts

```diff
@@ -13,31 +14,31 @@ export enum PortName {
 }
 
 export interface PortRequest {
-  [PortName.Completion]: { id: RequestId; request: CompletionRequest }
+  [PortName.Completion]: { id: RequestID; request: CompletionRequest }
   [PortName.Permission]: {
-    id?: RequestId
-    request: { requesterId: RequestId; permitted?: boolean }
+    id?: RequestID
+    request: { requesterId: RequestID; permitted?: boolean }
   }
-  [PortName.Model]: { id: RequestId; request: ModelRequest }
-  [PortName.Events]: { id?: RequestId; request: EventRequest<unknown> }
+  [PortName.Model]: { id: RequestID; request: ModelRequest }
+  [PortName.Events]: { id?: RequestID; request: EventRequest<unknown> }
 }
 
 export interface PortResponse {
```

Comment:
(Not relevant to this PR, but would love to hear your thoughts on how to do port request/response types sometime. Sticking them all in one file and not the `ports` files seems bad.)

Reply:
ref:

- https://github.com/PlasmoHQ/plasmo/issues/334
- https://docs.plasmo.com/framework/messaging#e2e-type-safety-wip

I'd say it depends - I found barrel patterns to be not flexible, but it's great to consolidate thought when planning out architecture. I generally do `import type {}` in the port consumer, and let the port files be the source of truth. The messaging/port API can incorporate better typing practice as well.

re: barrel https://flaming.codes/en/posts/barrel-files-in-javascript